homeassistant/components/modbus/entity.py

```diff
@@ -79,14 +80,19 @@ def __init__(
         """Initialize the Modbus binary sensor."""
 
         self._hub = hub
-        self._slave = entry.get(CONF_SLAVE) or entry.get(CONF_DEVICE_ADDRESS, 1)
+        self._slave = (
+            entry.get(CONF_SLAVE)
+            if entry.get(CONF_SLAVE) is not None
+            else entry.get(CONF_DEVICE_ADDRESS, 1)
+        )
```

Comment:
```suggestion
        if (conf_slave := entry.get(CONF_SLAVE)) is not None:
            self._slave = conf_slave
        else:
            self._slave = entry.get(CONF_DEVICE_ADDRESS, 1)
```

Usually we avoid a ternary when it spans multiple lines.